plugins/module_utils/identity/keycloak/keycloak.py

```diff
@@ -120,6 +123,75 @@ def __init__(self, module, connection_header):
         self.validate_certs = self.module.params.get('validate_certs')
         self.restheaders = connection_header
 
+    def get_realm_by_id(self, realm='master'):
+        """ Obtain realm representation by id
+
+        :param realm: realm id
+        :return: dict of real, representation or None if none matching exist
+        """
+        realm_url = URL_REALM.format(url=self.baseurl, realm=realm)
+
+        try:
+            return json.loads(to_native(open_url(realm_url, method='GET', headers=self.restheaders,
+                                                 validate_certs=self.validate_certs).read()))
+
+        except HTTPError as e:
+            if e.code == 404:
+                return None
+            else:
+                self.module.fail_json(msg='Could not obtain realm %s: %s'
+                                          % (realm, str(e)))
```

Comment:
@kris2kris it is a good practice to also pass the exception's traceback when calling `fail_json()`, as in:
```suggestion
                self.module.fail_json(msg='Could not obtain realm %s: %s' % (realm, str(e)),
                                      exception=traceback.format_exc())
```
Obviously applies to multiple places in this PR.